src/otx/core/data/module.py

```diff
@@ -0,0 +1,111 @@
+"""LightningDataModule extension for OTX."""
+from __future__ import annotations
+
+import logging as log
+from typing import TYPE_CHECKING
+
+from datumaro import Dataset as DmDataset
+from lightning import LightningDataModule
+from torch.utils.data import DataLoader
+
+from otx.core.types.task import OTXTaskType
+
+from .factory import OTXDatasetFactory
+
+if TYPE_CHECKING:
+    from otx.core.config.data import (
+        DataModuleConfig,
+        SubsetConfig,
+    )
+
+    from .dataset.base import OTXDataset
+
+
+class OTXDataModule(LightningDataModule):
+    """LightningDataModule extension for OTX pipeline."""
+
+    def __init__(self, task: OTXTaskType, config: DataModuleConfig) -> None:
+        """Constructor."""
+        super().__init__()
+        self.task = task
+        self.config = config
+        self.subsets: dict[str, OTXDataset] = {}
+        self.prepare_data_per_node = True
```

Comment:
GitHub Copilot: Where is `self.prepare_data_per_node` utilized within the code? And why is it required to be set to `True`? Isn't its default value already set to `True`?

Reply:
Removed here https://github.com/openvinotoolkit/training_extensions/pull/2670/commits/42527bef33ed641db38cd61f1759c34b4209a1f3